File: Programmers/Level1/220915/Lotto.js
export function highestNumber(lottos, win_nums) {
  const rightNumber = lottos.filter((number) => (
    win_nums.includes(number)
  ));

  const highestRank = {
    number: rightNumber.length,
  };

  lottos.forEach((number) => {
    if (number === 0) {
      highestRank.number += 1;
    }
  });

  return highestRank.number;
}

export function lowestNumber(lottos, win_nums) {
  const rightNumber = lottos.filter((number) => (
    win_nums.includes(number)
  ));

  return rightNumber.length;
}

export function lottoResult(highestRankNumber, lowestRankNumber) {
  const highestNumber = {
    rank: 7 - highestRankNumber,
  };

  if (highestRankNumber === 0) {
    highestNumber.rank = 6;
  }

  const lowestNumber = {
    rank: 7 - lowestRankNumber,
  };

  if (lowestRankNumber === 0) {
    lowestNumber.rank = 6;
  }

  return [highestNumber.rank, lowestNumber.rank];
}

export function solution(lottos, win_nums) {
  const highestRankNumber = highestNumber(lottos, win_nums);
  const lowestRankNumber = lowestNumber(lottos, win_nums);

  const answer = lottoResult(highestRankNumber, lowestRankNumber);

  return answer;
}

File: Programmers/Level1/220914/Java/Lotto.java
public class Lotto {
  public int[] solution(int[] lottos, int[] win_nums) {
    int[] answer;

    int highestRankCount = hopeRaffle(lottos, win_nums);
    int lowestRankCount = despairRaffle(lottos, win_nums);

    answer = rank(highestRankCount, lowestRankCount);

    return answer;
  }

  public int hopeRaffle(int[] lottos, int[] win_nums) {
    int rankCount = 0;

    for (int i = 0; i < lottos.length; i += 1) {
      for (int j = 0; j < win_nums.length; j += 1) {
        if (lottos[i] == 0) {
          rankCount += 1;
          break;
        }

        if (lottos[i] == win_nums[j]) {
          rankCount += 1;
        }
      }
    }
    return rankCount;
  }

  public int despairRaffle(int[] lottos, int[] win_nums) {
    int rankCount = 0;

    for (int i = 0; i < lottos.length; i += 1) {
      for (int j = 0; j < win_nums.length; j += 1) {
        if (lottos[i] == win_nums[j]) {
          rankCount += 1;
        }
      }
    }

    return rankCount;
  }

  public int[] rank(int highestRankCount, int lowestRankCount) {
    int rank[] = new int[2];

    switch (highestRankCount) {
      case 6 -> rank[0] = 1;
      case 5 -> rank[0] = 2;
      case 4 -> rank[0] = 3;
      case 3 -> rank[0] = 4;
      case 2 -> rank[0] = 5;
      default -> rank[0] = 6;
    }

    switch (lowestRankCount) {
      case 6 -> rank[1] = 1;
      case 5 -> rank[1] = 2;
      case 4 -> rank[1] = 3;
      case 3 -> rank[1] = 4;
      case 2 -> rank[1] = 5;
      default -> rank[1] = 6;
    }

    return rank;
  }
}
